Scripts/script-AssignAnalystToIncident.yml

```diff
@@ -28,37 +34,60 @@ script: |
+  }
+
   if (!userToAssign) {
-    switch(assignBy) {
-        case 'online':
-            var usersRes = executeCommand('getUsers', { roles: args.roles, online: true });
-            var usersList = usersRes[0].Contents.map(function (u) { return u.username });
-            userToAssign = usersList[Math.floor(Math.random() * usersList.length)];
-            break;
-        case 'current':
-          var usersRes = executeCommand('getUsers', { current: true });
-          var usersList = usersRes[0].Contents.map(function (u) { return u.username });
-          userToAssign = usersList[Math.floor(Math.random() * usersList.length)];
-          break;
-        case 'random':
-            var usersRes = executeCommand('getUsers', { roles: args.roles });
-            var usersList = usersRes[0].Contents.map(function (u) { return u.username });
-            userToAssign = usersList[Math.floor(Math.random() * usersList.length)];
-            break;
-        default:
-            res = executeCommand("getOwnerSuggestion", {})[0].Contents;
-            switch (assignBy) {
-                case 'machine-learning':
-                    userToAssign = res.ownerByMl;
-                    break;
-                case 'top-user':
-                    userToAssign = res.topOwner;
-                    break;
-                case 'less-busy-user':
-                    userToAssign = res.userLeastLoad;
-                    break;
-            }
-    }
+      switch(assignBy) {
+          case 'online':
+              var usersRes = executeCommand('getUsers', { roles: args.roles, online: true });
+              if (isError(usersRes[0])) {
+                  return getUserException;
+              } else {
+                  chooseUser(usersRes);
+              }
+              break;
+          case 'current':
+              var usersRes = executeCommand('getUsers', { current: true });
+              if (isError(usersRes[0])) {
+                  return getUserException;
+              } else {
+                  chooseUser(usersRes);
+              }
+              break;
+          case 'random':
+              var usersRes = executeCommand('getUsers', { roles: args.roles });
+              if (isError(usersRes[0])) {
+                  return getUserException;
+              } else {
+                  chooseUser(usersRes);
+              }
+              break;
+          default:
+              res = executeCommand("getOwnerSuggestion", {})[0].Contents;
+              switch (assignBy) {
+                  case 'machine-learning':
+                      userToAssign = res.ownerByMl;
+                      break;
+                  case 'top-user':
+                      userToAssign = res.topOwner;
+                      break;
+                  case 'less-busy-user':
+                      userToAssign = res.userLeastLoad;
+                      break;
+              }
+      }
+  }
+
+  if (!userToAssign) {
```

Comment:
Not sure about this part, if someone asks for online users and got empty response you might return here users which the caller is not expected.
It also breaks compatibility.

Reply:
You are right, moved it to the end of the default of the first switch. This way we will run this function only in case the getOwnerSuggestion command returned an empty value.